### PR TITLE
MAIN-29031 | Avoid infinite recursion when resetting IC cache

### PR DESCRIPTION
**Pull request by mszabo-wikia**
Infinite recursion may occur via resetCache() ->
Interpreter::getValues() -> SamplingCache when setting IC cache values. As a workaround, add a recursion guard till we figure out a proper fix.

### CODE REVIEW

**Comment by mszabo-wikia** [comment]
Likely flow: `InstantConfigCacheStorage::set` -> `InstantConfigCacheStorage::resetCache` -> `InstantConfigInterpreter::getValues` -> `SamplingCacheManager::apply` -> `InstantConfigCacheStorage::set`.

**Review by jakubbanas — APPROVED** [review]

**Review by latata — APPROVED** [review]

File: src/core/services/instant-config-cache-storage.ts [hunk 1]
```diff
@@ -35,7 +35,11 @@ export class InstantConfigCacheStorage implements InstantConfigCacheStorageServi
 	private readonly cacheKey = 'basset';
 
 	private constructor() {
-		this.resetCache();
+		communicationService.on(eventsRepository.AD_ENGINE_CONSENT_READY, ({ gdprConsent }) => {
```
Comment (mszabo-wikia) [inline src/core/services/instant-config-cache-storage.ts:38]: I think it's worth adding a comment around here, or on resetCache(), to be mindful of potential recursion like the one we're eliminating here, for the benefit of our future selves.

**Comment by jbienkowski-fandom** [comment]
Part 2: https://github.com/Wikia/identity-engine/pull/231

File: src/core/services/instant-config-cache-storage.ts (hunk 1 above)
Comment (jbienkowski-fandom) [reply src/core/services/instant-config-cache-storage.ts]: The whole flow got refactored a bit more to mitigate a chance of recursive issue as now we have a separate PostQuecast event for that

**Review by mfarul — APPROVED** [review]

**Pull request merged**